Platform/OcTextInputDxe/OcTextInputDxe.c

```diff
@@ -0,0 +1,867 @@
+          SuccessCount++;
+        } else if (Status != EFI_ALREADY_STARTED) {
+          DEBUG ((
+            DEBUG_WARN,
+            "OcTextInputDxe: Failed to install on handle %d (%p): %r\n",
+            Index,
+            HandleBuffer[Index],
+            Status
+            ));
+        }
+      }
+
+      gBS->FreePool (HandleBuffer);
+    } else {
+      DEBUG ((DEBUG_ERROR, "OcTextInputDxe: Failed to locate SimpleTextInput handles: %r\n", Status));
+    }
+  } else {
+    DEBUG ((
+      DEBUG_INFO,
+      "OcTextInputDxe: System appears to have full EFI 2.0+ support, compatibility layer not needed\n"
+      ));
+  }
+
+  gDriverInitialized = TRUE;
+  DEBUG ((
+    DEBUG_INFO,
+    "OcTextInputDxe: === Compatibility installation completed: %d successful installations ===\n",
+    SuccessCount
+    ));
+  return EFI_SUCCESS;
+}
+
+//
+// Event notification callback for ReadyToBoot
+//
+VOID
+EFIAPI
+OnReadyToBootEvent (
+  IN EFI_EVENT  Event,
+  IN VOID       *Context
+  )
+{
+  DEBUG ((
+    DEBUG_INFO,
+    "OcTextInputDxe: System is ready to boot, performing late initialization\n"
+    ));
+  PerformCompatibilityInstallation ();
+}
+
+/**
+   The entry point for OcTextInputDxe driver.
+
+   @param[in] ImageHandle    The firmware allocated handle for the EFI image.
+   @param[in] SystemTable    A pointer to the EFI System Table.
+
+   @retval EFI_SUCCESS       The entry point is executed successfully.
+   @retval other             Some error occurs when executing this entry point.
+ **/
+EFI_STATUS
+EFIAPI
+OcTextInputDxeEntry (
+  IN EFI_HANDLE        ImageHandle,
+  IN EFI_SYSTEM_TABLE  *SystemTable
+  )
+{
+  EFI_STATUS  Status;
+
+  DEBUG ((
+    DEBUG_INFO,
+    "OcTextInputDxe: === OpenCore SimpleTextInputEx Compatibility Driver Starting ===\n"
+    ));
+  DEBUG ((
+    DEBUG_INFO,
+    "OcTextInputDxe: System Table Revision: 0x%08X (%d.%02d)\n",
+    SystemTable->Hdr.Revision,
+    SystemTable->Hdr.Revision >> 16,
+    SystemTable->Hdr.Revision & 0xFFFF
+    ));
+
+  if (SystemTable->RuntimeServices != NULL) {
+    DEBUG ((
+      DEBUG_INFO,
+      "OcTextInputDxe: Runtime Services Revision: 0x%08X (%d.%02d)\n",
+      SystemTable->RuntimeServices->Hdr.Revision,
+      SystemTable->RuntimeServices->Hdr.Revision >> 16,
+      SystemTable->RuntimeServices->Hdr.Revision & 0xFFFF
+      ));
+  }
+
+  if (SystemTable->BootServices != NULL) {
+    DEBUG ((
+      DEBUG_INFO,
+      "OcTextInputDxe: Boot Services Revision: 0x%08X (%d.%02d)\n",
+      SystemTable->BootServices->Hdr.Revision,
+      SystemTable->BootServices->Hdr.Revision >> 16,
+      SystemTable->BootServices->Hdr.Revision & 0xFFFF
+      ));
+  }
+
+  // Check if console services are available for immediate installation
+  if ((SystemTable->ConsoleInHandle != NULL) && (SystemTable->ConIn != NULL)) {
+    DEBUG ((DEBUG_INFO, "OcTextInputDxe: Console services are available, attempting immediate installation\n"));
+    Status = PerformCompatibilityInstallation ();
```

Comment:
The implementation of PerformCompatibilityInstallation swallows it's errors and always returns EFI_SUCCESS.